src/v/s3/client.cc

```diff
@@ -725,6 +821,77 @@ ss::future<> client::delete_object(
       });
 }
 
+static auto iobuf_to_delete_objects_result(iobuf&& buf) {
+    auto root = iobuf_to_ptree(std::move(buf));
+    auto result = client::delete_objects_result{};
+    try {
+        for (auto const& [tag, value] : root.get_child("DeleteResult")) {
+            if (tag != "Error") {
+                continue;
+            }
+            auto code = value.get<std::string>(
+              "Code", "[no error code present]");
+            auto key = value.get<std::string>("Key", "[no key present]");
+            auto message = value.get<std::string>(
+              "Message", "[no error message present]");
+            auto version_id = value.get<std::string>(
+              "VersionId", "[no version id present]");
+            vlog(
+              s3_log.trace,
+              R"log(delete_objects_result::undeleted_keys Code:"{}" Key:"{}" Message:"{}" VersionId:"{}")log",
+              code,
+              key,
+              message,
+              version_id);
+            if (value.find("Key") == value.not_found()) {
+                vlog(
+                  s3_log.warn,
+                  "an DeleteResult.Error does not contain the Key tag");
+                continue;
+            }
+            result.undeleted_keys.push_back({
+              s3::object_key{value.get<std::string>("Key")},
+              value.get<std::string>("Code", "[no error code present]"),
+            });
+        }
+    } catch (...) {
+        vlog(s3_log.error, "!!error parse result {}", std::current_exception());
+        throw;
+    }
+    return result;
+}
+
+auto client::delete_objects(
+  bucket_name const& bucket,
+  std::span<const object_key> keys,
+  ss::lowres_clock::duration timeout) -> ss::future<delete_objects_result> {
+    auto request = _requestor.make_delete_objects_request(bucket, keys);
+    if (!request) {
+        return ss::make_exception_future<delete_objects_result>(
+          std::system_error(request.error()));
+    }
+    auto& [header, body] = request.value();
+    vlog(s3_log.trace, "send https request:\n{}", header);
+
+    return ss::do_with(
+             std::move(body),
```

Comment:
does this object have to be closed somewhere?

Reply:
what do you mean?

Reply:
It looks like `body` is a `ss::input_stream` which usually has to be closed after use. Is it closed inside the client request method? If not do we need to close it at the end in this method?

Reply:
Good question. I didn't get any error or warning in the unit tests but I'll check again the usage

Reply:
@abhijat in principle the caller is responsible to call ss::input_stream::closed, but since my ss::data_source_impl does not override it, there is no error.
But for correctness and future proofing I added a finally([&]{to_delete.close();})